### PR TITLE
Feature/create recipe repo

### LINKED ISSUE #1: RECIPE REPOSITORY - DATA
- [ ] add import & exports
- properties >>
- [ ] data = recipe data 
- methods >>
- [ ] a way to filter by tags > return a list
- [ ] a way to filter by name or ingredients
    > these are in user file?

### LINKED ISSUE #2: RECIPE REPOSITORY - TEST
- [ ] check import & export files
- [ ] test all properties
- [ ] test all methods

### PR DESCRIPTION
**Pull request by davidleach724**
# Description
This feature creates a RecipeRepository which takes in the recipeData. It incorporates 3 different methods: find RecipeTags, findTaggedRecipes, and filterSearchRecipes. It also includes a test file for the RecipeRepository.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [ ] open index.html
- [x] console.log()
- [ ] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -93,6 +93,8 @@ function addToDom(recipeInfo, shortRecipeName) {
 }
 
 // FILTER BY RECIPE TAGS - MOVE TO RecipeRepository.js
+// *** This was moved to RecipeRepo as method .findTags() that returns the sorted list of tags
```
Comment (Meggs625) [inline src/scripts.js:96]: Great note for when we refactor scripts

File: src/RecipeRepository.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      allRecipes.forEach(recipe => {
+        if (!filteredTagResults.includes(recipe)) {
+          filteredTagResults.push(recipe);
+        }
+      })
+    });
+   return filteredTagResults;
+  }
+
+  //might need to add functionality to lowercase all names/search material
```
Comment (madhaus4) [inline src/RecipeRepository.js:35]: agreed ..there is the function in the scripts file to do this so we can see how to make these talk to each other

File: src/scripts.js (hunk 1 above)
Comment (madhaus4) [inline src/scripts.js:96]: thanks for adding the comments to explain where new code location is

File: test/recipeRepository-test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+describe.only('RecipeRepository' , () => {
+  let recipeRepository;
+
+  beforeEach(() => {
+    recipeRepository = new RecipeRepository(recipeData);
+  })
+
+  it('should be a function', () => {
+    expect(RecipeRepository).to.be.a.a('function');
```
Comment (madhaus4) [inline test/recipeRepository-test.js]: there are 2 a's here fyi

File: test/recipeRepository-test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      'sauce',        'side dish',
+      'snack',        'spread',
+      'starter']);
+  })
+
+  it('should return recipes that include selected tags', () => {
+    expect(recipeRepository.findTaggedRecipes(['brunch']).length).to.equal(1)
+  })
+
+  it('should be able to search by name or ingredient', () => {
```
Comment (Meggs625) [inline test/recipeRepository-test.js:36]: how does Birdie feel about this ingredient list?

**Review by Meggs625 — APPROVED** [review]

File: test/recipeRepository-test.js (hunk 3 above)
Comment (davidleach724) [reply test/recipeRepository-test.js]: Thanks for the helpful catch!

File: test/recipeRepository-test.js (hunk 4 above)
Comment (davidleach724) [reply test/recipeRepository-test.js:36]: She is not amused.

**Pull request merged**